google/web-stories-wp
# Story Locking: Autosave on takeover

## Context



## Summary

Trigger an autosave when user has their story taken over.

## Relevant Technical Choices

 Only trigger if they are changes in the story. Otherwise, it will trigger a REST API call that returns a 400 error. 

## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:
0. Enable autosaving and take over post locking experiment.
1. Create a story. 
2. Add some content. 
3. Save as draft. 
4. Open another browser. 
5. Login, as a differnet user. 
6. Open story ( from previous ) in editor. 
7. Click Take over story. 
8. In first browser, make changes to story. DO NOT SAVE. 
9. Wait 1 minute 30 seconds. 
10. Go back to second browser. 
11. Refresh browser. 
12. See message that there is a newer autosave. 

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #12325
